cudamapper/src/main.cu

```diff
@@ -137,46 +150,44 @@ int main(int argc, char* argv[])
         std::vector<std::pair<std::int32_t, int32_t>> target_ranges;
     };
 
+    auto query_chunks  = query_parser->get_read_chunks(index_size * 1000000);
+    auto target_chunks = target_parser->get_read_chunks(target_index_size * 1000000);
+
     //First generate all the ranges independently, then loop over them.
     std::vector<query_target_range> query_target_ranges;
 
-    for (std::int32_t query_start_index = 0; query_start_index < queries; query_start_index += index_size)
-    {
-
-        std::int32_t query_end_index = std::min(query_start_index + index_size, queries);
-
-        query_target_range q;
-        q.query_range = std::make_pair(query_start_index, query_end_index);
+    // TODO: here goes filling out the query_target_ranges vector but using chunks
 
-        std::int32_t target_start_index = 0;
-        // If all_to_all mode, then we can optimzie by starting the target sequences from the same index as
-        // query because all indices before the current query index are guaranteed to have been processed in
-        // a2a mapping.
-        if (all_to_all)
+    int target_idx = 0;
+    for (auto query_chunk : query_chunks)
+    {
+        query_target_range range;
+        range.query_range = query_chunk;
+        for (int t = target_idx; t < target_chunks.size(); t++)
```

Comment:
Is this comment in the correct location? This loop does not use strings.

Reply:
Looks like it's not :man_facepalming: Let's see if I remember what it was about when doing the rereview

Reply:
Ok. I think I know what I wanted to say here (no idea how `string` got into play though).
Instead of looping over all targets explicitly you could do something like:
`std::copy(std::begin(target_chunks) + offset_for_all_to_alll, std::end(target_chunks), std::back_inserter(range.target_ranges))`
This might or might not be more readable than the explicit loop